docs/en/cpp/guide/mavlink_direct.md

```diff
@@ -0,0 +1,226 @@
+    ],
+    "increment": 0,
+    "min_distance": 000,
+    "max_distance": 2000,
+    "increment_f": 10.0,
+    "angle_offset": -40.0,
+    "frame": 12
+})";
+
+while (true) {
+    // Send the message
+    auto result = mavlink_direct.send_message(obstacle_distance);
+    if (result == MavlinkDirect::Result::Success) {
+        std::cout << "OBSTACLE_DISTANCE message sent successfully" << std::endl;
+    } else {
+        std::cerr << "OBSTACLE_DISTANCE message could not be sent: " << result << std::endl;
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+}
+```
+
+Check the full example on [GitHub](https://github.com/mavlink/MAVSDK/tree/main/examples/mavlink_direct_sender).
+
+### Receiving Messages
+
+#### Subscribe to Specific Message Types
+
+```cpp
+// Subscribe to GLOBAL_POSITION_INT messages
+auto handle = mavlink_direct.subscribe_message(
+    "GLOBAL_POSITION_INT",
+    [](MavlinkDirect::MavlinkMessage message) {
+        std::cout << "Received position: " << message.fields_json << std::endl;
+
+        // And this could now be parsed by jsoncpp or nlohmann/json:
+        // Json::Value json;
+        // Json::Reader reader;
+        // if (reader.parse(message.fields_json, json)) {
+        //     auto lat = json["lat"].asInt() / 1e7;  // Convert from degrees * 1e7
+        //     auto lon = json["lon"].asInt() / 1e7;
+        //     std::cout << "Position: " << lat << ", " << lon << std::endl;
+        // }
+    }
+);
+```
+
+#### Subscribe to All Messages
+
+```cpp
+// Subscribe to all incoming messages
+auto handle = mavlink_direct.subscribe_message(
+    "",  // Empty string = all messages
+    [](MavlinkDirect::MavlinkMessage message) {
+        std::cout << "Received " << message.message_name
+                  << " from system " << message.system_id
+                  << " component " << message.component_id << std::endl;
+    }
+);
+```
+
+#### Unsubscribe
+
+```cpp
+// Clean up subscription
+mavlink_direct.unsubscribe_message(handle);
+```
+
+### Sniffing all messages
+
+There is also [an example that demonstrates a generic MAVLink message sniffer](https://github.com/mavlink/MAVSDK/tree/main/examples/sniffer).
+
+::: info
+: This example uses the mavsdk.h intercept API rather than the MavlinkDirect plugin API, however, they are very similar.
+:::
+
+## JSON Field Format
+
+MavlinkDirect represents MAVLink message fields as JSON objects:
+
+```json
+{
+  "time_boot_ms": 12345,
+  "lat": 473977418,
+  "lon": -1223974560,
+  "alt": 100500,
+  "relative_alt": 50250,
+  "vx": 100,
+  "vy": -50,
+  "vz": 25,
+  "hdg": 18000
+}
+```
+
+**Key points:**
+
+- All field names match the MAVLink message definition
+- Numeric values preserve original MAVLink types and scaling
+- Arrays are represented as JSON arrays: `[1, 2, 3, 0, 0]`
+- NaN and infinity values are represented as `null`
+- Extended fields are automatically included
+
+## Load custom MAVLink xml / MAVLink dialects
```

Comment:
Ah, so I think this needs to be near the top of the page with the comment that MAVSDK already loads X, Y, Z, This is how you load additional stuff.

Reply:
I've moved it up.